app/lib/menu.rb

```diff
@@ -13,11 +15,74 @@ def initialize
     @items = {}
   end
 
+  # @param name [String]
+  # @param svg [String]
+  # @param children [Array<Hash>]
   def scope(name, svg, children)
-    @items[name] = { svg: "#{svg}.svg", children: children }
+    @items[name] = Scope.new(name: name, svg: "#{svg}.svg", children: children)
+  end
+
+  # @see {Item#initialize} for details on the parameters.
+  # @return [Menu::Item]
+  def item(...)
+    Item.new(...)
   end
 
-  def item(name:, controller: name, children: [], visible: true, parent: nil)
-    { name: name, controller: controller.tr(" ", "_"), children: children, visible: visible, parent: parent }
+  class Scope
+    attr_reader :name, :svg, :children
+
+    def initialize(name:, svg:, children:)
+      @name = name
+      @svg = svg
+      @children = children
+    end
+
+    # Does this Scope have multiple children?  From a presentation stand point
+    # we render menu items with multiple children a bit differently than those
+    # with only one child.
```

Comment:
Do you think that this comment is necessary as its implied from the view? Perhaps we could rather document the method if you think it needs further docs.

Reply:
I was surprised when I saw a method helper that was "children?" and it was "children.count > 1".  So I'm adding this as a thing to remember.